eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/FluentMethodNameCheck.java

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Model Class Method:
+ *  Fluent Methods: All methods that return an instance of the class, and with one parameter.
+ *  The method name should not start with {@code avoidStartWord}.
+ */
+public class FluentMethodNameCheck extends AbstractCheck {
+
+    private static final String FLUENT_METHOD_ERR = "'%s' fluent method name should not start with keyword '%s'.";
+
+    // Specifies valid identifier: default start word is 'with', modelName is model
+    private String[] avoidStartWords = new String[] {"with"};   // by default
+    private String modelName = "model";                         // by default
+
+    private static final String DOT = ".";
+    private String className;
+    private static boolean isModelClass;
+
+    @Override
+    public void beginTree(DetailAST ast) {
+        this.isModelClass = false;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.METHOD_DEF,
+            TokenTypes.PACKAGE_DEF
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.PACKAGE_DEF:
+                FullIdent packageNameFI = FullIdent.createFullIdentBelow(ast);
+                this.isModelClass = packageNameFI.getText().endsWith(DOT + modelName);
```

Comment:
This isn't an accurate enough way to determine a model class (all classes that are in a package whose name ends with `.model`). There may be sub-packages, or the class may not even be in a `.model` package at all.

Reply:
Is sufficient enough to determine a class is a model class by checking all properties in the class has getter and setter methods?

Reply:
No, I wouldn't say that this is sufficient. I think a good starting point is to identify any methods that return an instance of the same class. This should identify most fluent methods, and we can augment that at a later point if necessary.

Reply:
So we don't need to check if the class is model class. We only check for fluent method now?

Reply:
There is no standard definition of what a model class is - looking to find something that confers a class is a model class will be very error prone. What I'm suggesting is finding what appear to be fluent setter methods and making sure they match a pattern.